core/src/main/java/org/apache/iceberg/SnapshotProducer.java

```diff
@@ -615,4 +633,47 @@ private static void updateTotal(
       }
     }
   }
+
+  private Long snapshotIdToRollbackToOnConflict(Long parentSnapshotId) {
```

Comment:
can we add a javdoc for this to explain the algorithm instead of embedding all the comments inline? Also the method name feels like just getting the ID, but in fact you are already performing the rollback. So maybe an actionable method name like just `rollbackReplaceOnConflictToParent(Long originalParent)`?

Reply:
we are not actually doing a rollback here, but rather trying to see which parent snapshot id we can rollback to, once we figure that out we apply the current on this parent snapshot id we found and commit to the table so the actual rollback is happening in `apply()` hence named this method as it is, please let me know your thoughts, happy to make the changes accordingly